src/app/shared/components/quantity-input/quantity-input.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, Input } from "@angular/core";
+import { AbstractControl, ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR, ValidationErrors, Validator } from "@angular/forms";
+
+@Component({
+    selector: "app-quantity-input",
+    templateUrl: "./quantity-input.component.html",
+    styleUrls: ["./quantity-input.component.scss"],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: QuantityInputComponent,
+        },
+        {
+            provide: NG_VALIDATORS,
+            multi: true,
+            useExisting: QuantityInputComponent,
+        }
+    ]
+})
+export class QuantityInputComponent implements ControlValueAccessor, Validator {
+    @Input() label: string = "";
+    @Input() maxValue?: number;
```

Comment:
I like the optional input, but it would make more sense to define default values that define the base behaviour we want. So here, maybe max is 20 by default, and minValue should be 0